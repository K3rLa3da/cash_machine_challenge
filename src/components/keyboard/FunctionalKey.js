import React from 'react';

const FunctionalKey = props => {
    const {button: {name, color}, onButtonClick} = props;
    return (
        <div className="functional-key-container">
            <div className={`keyboard-key ${color}`}
                 onClick={onButtonClick}
            >
                {name}
            </div>
        </div>
    );
};

export default FunctionalKey;