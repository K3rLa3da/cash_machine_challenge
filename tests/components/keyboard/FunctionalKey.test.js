import React from 'react';
import {shallow} from 'enzyme';
import FunctionalKey from '../../../src/components/keyboard/FunctionalKey';


test('test render FunctionalKey', () => {
    const mockButtonClick = jest.fn();
    const props = {
        button: {name: 'Accept', color: 'green'},
        onButtonClick: mockButtonClick,
    };
    const wrapper = shallow(
        <FunctionalKey {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
    const key = wrapper.find('.keyboard-key');
    expect(key.hasClass('green')).toBeTruthy();
    expect(key.text()).toBe('Accept');
    key.simulate('click');
    expect(mockButtonClick).toHaveBeenCalled();
});
