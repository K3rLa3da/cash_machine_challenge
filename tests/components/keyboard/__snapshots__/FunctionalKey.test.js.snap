// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render FunctionalKey 1`] = `
<div
  className="functional-key-container"
>
  <div
    className="keyboard-key green"
    onClick={[MockFunction]}
  >
    Accept
  </div>
</div>
`;
